search/search_service/proxy/es_search_proxy.py

```diff
@@ -0,0 +1,313 @@
+from elasticsearch_dsl.query import MultiMatch
+from elasticsearch_dsl.response import Response
+
+
+BOOL_QUERY = 'bool'
+WILDCARD_QUERY = 'wildcard'
+TERM_QUERY = 'term'
+TERMS_QUERY = 'terms'
+
+
+class Resource(Enum):
+    TABLE = 0
+    DASHBOARD = 1
+    FEATURE = 2
+    USER = 3
+
+
+RESOURCE_STR_MAPPING = {
+    'table': Resource.TABLE,
+    'dashboard': Resource.DASHBOARD,
+    'feature': Resource.FEATURE,
+    'user': Resource.USER,
+}
+
+
+class Filter():
+    def __init__(self, name: str, values: list, operation: str) -> None:
+        self.name = name
+        self.values = values
+        self.operation = operation
+
+
+class ElasticsearchProxy():
+    PRIMARY_ENTITIES = [Resource.TABLE, Resource.DASHBOARD, Resource.FEATURE, Resource.USER]
+
+    # mapping to translate request for table resources
+    TABLE_MAPPING = {
+        'key': 'key',
+        'badges': 'badges',
+        'tag': 'tags',
+        'schema': 'schema.raw',
+        'table': 'name.raw',
+        'column': 'column_names.raw',
+        'database': 'database.raw',
+        'cluster': 'cluster.raw'
+    }
+
+    # mapping to translate request for dashboard resources
+    DASHBOARD_MAPPING = {
+        'group_name': 'group_name.raw',
+        'name': 'name.raw',
+        'product': 'product',
+        'tag': 'tags',
+    }
+
+    # mapping to translate request for feature resources
+    FEATURE_MAPPING = {
+        'key': 'key',
+        'feature_group': 'feature_group.raw',
+        'feature_name': 'feature_name.raw',
+        'entity': 'entity',
+        'status': 'status',
+        'version': 'version',
+        'availability': 'availability.raw',
+        'tags': 'tags',
+        'badges': 'badges'
+    }
+
+    USER_MAPPING = {
+        'full_name': 'full_name',
+        'first_name': 'first_name',
+        'last_name': 'last_name',
+        'email': 'email'
+    }
+
+    RESOUCE_TO_MAPPING = {
+        Resource.TABLE: TABLE_MAPPING,
+        Resource.DASHBOARD: DASHBOARD_MAPPING,
+        Resource.FEATURE: FEATURE_MAPPING,
+        Resource.USER: USER_MAPPING,
+    }
+
+    def __init__(self, *,
+                 host: str = None,
+                 user: str = '',
+                 password: str = '',
+                 client: Elasticsearch = None) -> None:
+        # TODO actually implement this
+        if client:
+            self.elasticsearch = client
+        else:
+            http_auth = (user, password) if user else None
+            # doesn't this go against the whole point oh having a singleton pattern?
+            self.elasticsearch = Elasticsearch(host, http_auth=http_auth)
+
+
+    def _build_term_query(self, resource: Resource, query_term: str) -> Q:
+        """
+        Builds the query object for the inputed search term
+        """
+        if query_term == "" or not query_term:
+            # We don't want to create multi_match query for ""
+            # because it will result in no matches even with filters
+            return None
```

Comment:
if we allow returning None then return type of this function should be Optional[None]